### ISSUE TITLE
Capturing of grid-based one-electron properties fails

### ISSUE THREAD

**Issue by jwoisess** [psi4/psi4#1614]
I am aiming at the grid-based evaluation of molecular electrostatic fields/potentials as depicted in the input example [props4](https://github.com/psi4/psi4/blob/master/samples/props4/input.dat).

Running Psi4 1.1, the input example fails with the message
`AttributeError: 'builtin_function_or_method' object has no attribute 'Vvals'`.
Calling `Exvals`, `Eyvals` or `Ezvals` results in the same error.

Furthermore, I experienced the saving of `grid_field.dat` and `grid_esp.dat`, which follows after calling `prop()`, to be extremely slow (~1 MB per 5 minutes).

In conclusion, I kindly ask for responses to the following questions:
- What is going wrong with the direct capturing of one-electron properties as Python arrays?
- How could the saving of the `.dat` files be sped up?
- Is there a way to change the output file names to something different than `grid_field.dat`/ `grid_esp.dat`?

**Comment by susilehtola** [comment]
At least the speed issue is probably fundamental. Calculating the potential and the field at a grid point amounts to calculating a full set of one-electron Coulomb integrals for every one of these points. (Physically, the potential caused by a distribution of electron density onto a point r is the same as evaluating the effect of a charged nucleus at r onto the distribution of electrons.)

**Comment by jwoisess** [comment]
Are you saying it's the actual evaluation of the electrostatic field that determines the speed? I attempted to throw plenty of memory and dozens of threads at the problem. But the rate, at which the `.dat` files are written, remains exactly the same.

**Comment by susilehtola** [comment]
Yes, that's what I'm saying - you're basically evaluating the one-electron integrals five million times. What is your system?

**Comment by jwoisess** [comment]
I am running Psi4 on a Linux-based computing cluster with individual nodes that house 64 CPUs and 512 GB of memory. So the one-electron properties aren't evaluated on a grid but rather at single points while looping over all grid entries?

**Comment by susilehtola** [comment]
That would be my guess, but someone else can probably give a more accurate answer. I don't think evaluating millions of points is the intended use case for that code.

If you really do need that many points, then instead of analytic integrals, grid-based methods would likely be a lot faster.

**Comment by hokru** [comment]
A quick look suggest little to no use is made of threading:
https://github.com/psi4/psi4/blob/master/psi4/src/psi4/libmints/oeprop.cc#L980

With typical (=coarse) cubic grids it should be fast enough.

**Comment by susilehtola** [comment]
That depends on whether `integral_->electrostatic()` is threaded or not. But threading over grid points would probably be faster in this case.

**Comment by JonathonMisiewicz** [comment]
The way I read this topic, the action item here is to thread `grid_field` and `grid_esp` in oeprop.
